src/provider.rs

```diff
@@ -44,18 +44,56 @@ where
     }
 }
 
+#[async_trait]
+impl Middleware for Box<dyn Middleware  + 'static>
```

Comment:
is there a reason the static bound is needed?

Reply:
Oh, you're right, that shouldn't be necessary, I guess I assumed it was.